servicenow/getlogs_servicenow-3.8.py

```diff
@@ -0,0 +1,1644 @@
+        start_time = agent_config_vars['start_time']
+        end_time = agent_config_vars['end_time']
+        statement = agent_config_vars['timestamp_field'][
+                        0] + 'BETWEENjavascript:gs.dateGenerate(\'<START_DATE>\',\'00:00:00\')@javascript:gs.dateGenerate(\'<END_DATE>\',\'23:59:59\')'
+        statement = statement.replace('<START_DATE>', start_time).replace('<END_DATE>', end_time)
+        passthru['sysparm_query'] = '{}^OR{}'.format(passthru['sysparm_query'], statement) if len(
+            passthru['sysparm_query']) != 0 else statement
+    elif agent_config_vars['is_historical'] == False:
+        if agent_config_vars['cron_start_time']:
+            utc_cron_epoch = agent_config_vars['cron_start_time']
+            # get localized earliest datetime
+            local_cron_datetime = datetime.fromtimestamp(float(utc_cron_epoch))
+            # convert earliest datetime to the data timezone
+            data_cron_datetime = local_cron_datetime.astimezone(
+                agent_config_vars['timezone'])
+            # do not apply timezone conversion later
+            agent_config_vars['timezone'] = pytz.utc
+            # convert to string for Glide
+            cron_date_and_time = data_cron_datetime.strftime(
+                agent_config_vars['timestamp_format'][0]).split(' ')
+            earliest_date = cron_date_and_time[0]
+            earliest_time = cron_date_and_time[1]
+        else:
+            earliest_date = earliest_date_and_time[0]
+            earliest_time = '00:00:00'
+        for timestamp_field in agent_config_vars['timestamp_field']:
+            if not is_formatted(timestamp_field):
+                statement = '{}>=javascript:gs.dateGenerate(\'{}\',\'{}\')'.format(
+                    timestamp_field,
+                    earliest_date,
+                    earliest_time)
+                # OR between fields
+                passthru['sysparm_query'] = '{}^OR{}'.format(passthru['sysparm_query'], statement) if len(
+                    passthru['sysparm_query']) != 0 else statement
+        passthru['sysparm_offset'] = 0
+    # add applicable keyword filtering
+    for in_filter in agent_config_vars['filters_include']:
+        filter_keyword, filter_values = in_filter.split(':')
+        filter_statement = ''
+        for filter_value in filter_values.split(','):
+            statement = '{keyword}*{value}^{keyword}ISNOTEMPTY'.format(keyword=filter_keyword, value=filter_value)
+            # OR between values
+            filter_statement = '{}^OR{}'.format(filter_statement, statement) if len(
+                filter_statement) != 0 else statement
+        # AND between keywords
+        passthru['sysparm_query'] = '{}^{}'.format(passthru['sysparm_query'], filter_statement) if len(
+            passthru['sysparm_query']) != 0 else '{}'.format(filter_statement)
+    # clear since it's handled already
+    agent_config_vars['filters_include'] = ''
```

Comment:
Do we need to clear this variable? Typically, the agent_config_vars is a static variable reflecting the configuration in the config.ini.